packages/grid/_modules_/grid/hooks/features/treeData/useGridTreeData.ts

```diff
@@ -0,0 +1,149 @@
+import * as React from 'react';
+import { GridApiRef } from '../../../models/api/gridApiRef';
+import { GridComponentProps } from '../../../GridComponentProps';
+import { GRID_TREE_DATA_GROUP_COL_DEF } from './gridTreeDataGroupColDef';
+import { useGridApiEventHandler } from '../../utils/useGridApiEventHandler';
+import { GridEvents } from '../../../constants';
+import {
+  GridCellParams,
+  GridColDef,
+  GridColDefOverrideParams,
+  GridColumns,
+  MuiEvent,
+} from '../../../models';
+import { isSpaceKey } from '../../../utils/keyboardUtils';
+import { useFirstRender } from '../../utils/useFirstRender';
+import { buildRowTree } from '../../../utils/rowTreeUtils';
+import { GridRowGroupingPreProcessing } from '../../core/rowGroupsPerProcessing';
+import { isFunction } from '../../../utils/utils';
+import { gridFilteredDescendantCountLookupSelector } from '../filter';
+import { GridPreProcessingGroup } from '../../core/preProcessing';
+
+/**
+ * Only available in DataGridPro
+ * @requires useGridPreProcessing (method)
+ * @requires useGridRowGroupsPreProcessing (method)
+ */
+export const useGridTreeData = (
+  apiRef: GridApiRef,
+  props: Pick<
+    GridComponentProps,
+    'treeData' | 'getTreeDataPath' | 'groupingColDef' | 'defaultGroupingExpansionDepth'
+  >,
+) => {
+  const groupingColDef = React.useMemo<GridColDef>(() => {
+    const propGroupingColDef = props.groupingColDef;
+
+    const baseColDef: GridColDef = {
+      ...GRID_TREE_DATA_GROUP_COL_DEF,
+      headerName: apiRef.current.getLocaleText('treeDataGroupingHeaderName'),
+    };
+    let colDefOverride: Partial<GridColDef>;
+
+    if (isFunction(propGroupingColDef)) {
```

Comment:
The idea is to remove these one-line helpers.

```suggestion
    if (typeof propGroupingColDef === 'function') {
```